cdc/sink/dispatcher/event_router.go

```diff
@@ -0,0 +1,164 @@
+		log.Warn("can't support dispatch rule, using default rule", zap.String("rule", rule))
+	}
+}
+
+// EventRouter is a router, it determines which topic and which partition an event should be dispatched to.
+type EventRouter struct {
+	rules []struct {
+		partitionDispatcher partition.Dispatcher
+		topicDispatcher     topic.Dispatcher
+		filter.Filter
+	}
+}
+
+// DispatchRowChangedEvent returns the target topic and partition.
+func (s *EventRouter) DispatchRowChangedEvent(row *model.RowChangedEvent) (string, int32) {
+	topicDispatcher, partitionDispatcher := s.matchDispatcher(row)
+	topic := topicDispatcher.DispatchRowChangedEvent(row)
+	partition := partitionDispatcher.DispatchRowChangedEvent(row)
+
+	return topic, partition
+}
+
+// matchDispatcher returns the target topic dispatcher and partition dispatcher if a
+// row changed event matches a specific table filter.
+func (s *EventRouter) matchDispatcher(row *model.RowChangedEvent) (topic.Dispatcher, partition.Dispatcher) {
+	for _, rule := range s.rules {
+		if !rule.MatchTable(row.Table.Schema, row.Table.Table) {
+			continue
+		}
+		return rule.topicDispatcher, rule.partitionDispatcher
+	}
+	log.Panic("the dispatch rule must cover all tables")
+	return nil, nil
+}
+
+// getPartitionDispatcher returns the partition dispatcher for a specific partition rule.
+func getPartitionDispatcher(ruleConfig *config.DispatchRule, partitionNum int32, enableOldValue bool) partition.Dispatcher {
+	var (
+		d    partition.Dispatcher
+		rule partitionDispatchRule
+	)
+	rule.fromString(ruleConfig.PartitionRule)
+	switch rule {
+	case partitionDispatchRuleRowID, partitionDispatchRuleIndexValue:
+		if enableOldValue {
+			log.Warn("This index-value distribution mode " +
+				"does not guarantee row-level orderliness when " +
+				"switching on the old value, so please use caution!")
+		}
+		d = partition.NewIndexValueDispatcher(partitionNum)
+	case partitionDispatchRuleTS:
+		d = partition.NewTsDispatcher(partitionNum)
+	case partitionDispatchRuleTable:
+		d = partition.NewTableDispatcher(partitionNum)
+	case partitionDispatchRuleDefault:
+		d = partition.NewDefaultDispatcher(partitionNum, enableOldValue)
+	}
+
+	return d
+}
+
+// getTopicDispatcher returns the topic dispatcher for a specific topic rule (aka topic expression).
+func getTopicDispatcher(ruleConfig *config.DispatchRule, defaultTopic string) (topic.Dispatcher, error) {
+	if ruleConfig.TopicRule == "" {
+		return topic.NewStaticTopicDispatcher(defaultTopic), nil
+	}
+
+	// check if this rule is a valid topic expression
+	topicExpr := topic.Expression(ruleConfig.TopicRule)
+	err := topicExpr.Validate()
+	if err != nil {
+		return nil, err
+	}
+	return topic.NewDynamicTopicDispatcher(defaultTopic, topicExpr), nil
+}
+
+// NewEventRouter creates a new EventRouter
+func NewEventRouter(cfg *config.ReplicaConfig, partitionNum int32, defaultTopic string) (*EventRouter, error) {
+	ruleConfigs := append(cfg.Sink.DispatchRules, &config.DispatchRule{
```

Comment:
Do we maybe need to bring up this particular rule separately? Or at least add some notes.

Reply:
Done.